src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -87,6 +88,234 @@ protected override Conversion GetInterpolatedStringConversion(BoundUnconvertedIn
+                if (part is BoundStringInsert insert)
+                {
+                    if (interpolationHoleCandidates is null && !tryGetCandidateMethods("TryFormatInterpolationHole", ref useSiteInfo, out interpolationHoleCandidates))
+                    {
+                        // PROTOTYPE(interp-string): We'll likely want to continue attempting to bind for errors when we're not directly being converted to a string
+                        free();
+                        builderFormatCalls.Free();
+                        builderArguments = default;
+                        return false;
+                    }
+
+                    candidateMethods = interpolationHoleCandidates;
+                    analyzedArguments.Arguments.Add(insert.Value);
+                    analyzedArguments.Names.Add(null);
+
+                    if (insert.Alignment is not null)
+                    {
+                        analyzedArguments.Arguments.Add(insert.Alignment);
+                        analyzedArguments.Names.Add(SyntaxFactory.IdentifierName("alignment"));
+                    }
+                    if (insert.Format is not null)
+                    {
+                        analyzedArguments.Arguments.Add(insert.Format);
+                        analyzedArguments.Names.Add(SyntaxFactory.IdentifierName("format"));
+                    }
+                }
+                else
+                {
+                    candidateMethods = baseStringCandidates;
+                    analyzedArguments.Arguments.Add(part);
+                }
+
+                _binder.OverloadResolution.MethodInvocationOverloadResolution(
+                    candidateMethods,
+                    typeArguments,
+                    implicitBuilderReceiver,
+                    analyzedArguments,
+                    overloadResolutionResult,
+                    ref useSiteInfo);
+
+                if (!overloadResolutionResult.Succeeded
+                    || overloadResolutionResult.ValidResult.Member.CallsAreOmitted(source.SyntaxTree)
+                    || !returnTypeMatches(overloadResolutionResult.ValidResult.Member, usesBoolReturn))
+                {
+                    // PROTOTYPE(interp-string): We'll likely want to continue attempting to bind for errors when we're not directly being converted to a string
+                    free();
+                    builderFormatCalls.Free();
+                    interpolationHoleCandidates?.Free();
+                    builderArguments = default;
+                    return false;
+                }
+
+                var argsToParam = overloadResolutionResult.ValidResult.Result.ArgsToParamsOpt;
+
+                _binder.CoerceArguments(overloadResolutionResult.ValidResult, analyzedArguments.Arguments, diagnostics);
+
+                bool expanded = overloadResolutionResult.ValidResult.Result.Kind == MemberResolutionKind.ApplicableInExpandedForm;
```

Comment:
So the spec above says 'overload resolution contains a single result'... which matches here. I guess I'm suprised that we're allowing the candidate methods to include params and default args, etc, given that we're fairly prescriptive about the rest of the shape. Is there an explicit use case for this?

Reply:
> I guess I'm suprised that we're allowing the candidate methods to include params and default args, etc, given that we're fairly prescriptive about the rest of the shape.

We're not that prescriptive. I've taken a couple of shortcuts because of the specific type involved with interpolated strings as strings, but in general we'll have this type of support for `Create` and `Dispose` (if we support dispose at all).

---
In reply to: [609090339](https://github.com/dotnet/roslyn/pull/52422#discussion_r609090339) [](ancestors = 609090339)

Reply:
As to use cases: any of the caller info attributes, for example.

---
In reply to: [609114486](https://github.com/dotnet/roslyn/pull/52422#discussion_r609114486) [](ancestors = 609114486,609090339)